Chat fix up

Jackdehaan opened the pull request:
Improved the chat:

    Deleted unesasery lines and files
    Replaced axios for swagger api

Jackdehaan committed:
Started implementing Me user request

Jackdehaan committed:
Try to add chatroomsResponse but does not appear in generated-api

Jackdehaan committed:
Added custom hooks for userDetails and AddParticipant

Jackdehaan committed:
Changed PostChatroom to api call and moved it to utils -> chatroom
Only issue is that after creation the owner is not added as participant

Jackdehaan committed:
Added messege api call that causes : Authentication issue

Jackdehaan committed:
added useActiveParticipantbyChatroomId not working yet

Jackdehaan committed:
Put request and deleted not used code

Jackdehaan committed:
Merge branch 'develop' into chat-fix-up

Jackdehaan committed:
Merge branch 'develop' into chat-fix-up

Jackdehaan committed:
Merge develop into chat-fix

Jackdehaan committed:
Merge #94 develop into chat-fix

Jackdehaan committed:
Enforced prettier with npx prettier --write .

Jackdehaan committed:
Removed some eslint errors

rubenfin commented on frontend/src/utils/PostRequest.tsx:
Can you remove this file if it's unused

Jackdehaan committed:
Deleted files that where unused

Jackdehaan committed:
Added wrongly deleted file and fixed the not user found when uploading a message

daanmlab commented on frontend/src/App.tsx:
```diff
@@ -47,16 +48,15 @@ function App() {
-            <Route path="/" element={<Home />} />
-            <Route path="/update" element={<UpdateUser />} />
-            <Route path="/profile" element={<UserProfile />} />
-            <Route path="/profile/:id" element={<VisitingProfile />} />
```
Could you put this back svp

daanmlab commented on frontend/src/chat/ChatApiCalls.tsx:
Can you change this to a ts file?

daanmlab commented on frontend/src/chatroom/ApiRequest.tsx:
This can also be a ts file

daanmlab commented on frontend/src/chat/ChatApiCalls.tsx:
```diff
@@ -0,0 +1,140 @@
+import { useApi } from "@/utils/api/index.ts";
+import { useEffect, useState } from "react";
+
+export const useUserDetails = () => {
```
please remove this and use the existing userProvider like in Home.tsx:18

daanmlab reviewed (changes requested):
Nice job, some changes are needed but besides that this looks great!

Jackdehaan committed:
Fixed issues commented by Daan
- Put back the routes in app.tsx
- Changed chatapicalls and apirequest to a ts file
- Removed function useUserDetails and used the userProvider instead

Jackdehaan committed:
Merge branch 'develop' into chat-fix-up

Jackdehaan committed:
Fixed eslint issues

Jackdehaan committed:
Merge branch 'chat-fix-up' of https://github.com/eunbi0308/Transcendence into chat-fix-up

daanmlab reviewed (approved):
Niceee
